apis/ecr/v1alpha1/repository_types.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// RepositoryParameters define the desired state of an AWS Elastic Container Repository
+type RepositoryParameters struct {
+
+	// The image scanning configuration for the repository. This determines whether
+	// images are scanned for known vulnerabilities after being pushed to the repository.
+	// +optional
+	ImageScanningConfiguration *ImageScanningConfiguration `json:"imageScanningConfiguration,omitempty"`
+
+	// The tag mutability setting for the repository. If this parameter is omitted,
+	// the default setting of MUTABLE will be used which will allow image tags to
+	// be overwritten. If IMMUTABLE is specified, all image tags within the repository
+	// will be immutable which will prevent them from being overwritten.
+	// +optional
+	// +kubebuilder:validation:Enum=MUTABLE;IMMUTABLE
+	ImageTagMutability *string `json:"imageTagMutability,omitempty"`
+
+	// Metadata tagging key value pairs
+	// +optional
+	Tags []Tag `json:"tags,omitempty"`
+}
+
+// A RepositorySpec defines the desired state of a Elastic Container Repository.
+type RepositorySpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	// +optional
+	ForProvider RepositoryParameters `json:"forProvider"`
+}
+
+// RepositoryObservation keeps the state for the external resource
+type RepositoryObservation struct {
+	// The date and time, in JavaScript date format, when the repository was created.
+	CreatedAt *metav1.Time `json:"createdAt,omitempty"`
+
+	// The image scanning configuration for a repository.
+	ImageScanningConfiguration ImageScanningConfiguration `json:"imageScanningConfiguration,omitempty"`
```

Comment:
Can we omit and late initialize in the spec?

Reply:
Not sure what you mean on this one. Have it being set in late initialize already if left out of parameters. Thought all the properties should be in the observation?

Reply:
I just meant that I don't think we need it present in the observation if it will always match the spec exactly

Reply:
Oh gotcha. I was looking at it as this could actually be different in the spec vs what is actually in aws as you can update this field in the spec. So there is that brief period where they could be different before update runs. But if thats not a concern then yeah I can pull it.

Reply:
Hmm yeah I think generally we would say that if the field in the spec is not accurate then the resource should indicate that it is updating to match that state, but not necessarily that the current state of the field is reflected in the status. I think we could also remove `ImageTagMutability` from the status below. More info [here](https://github.com/crossplane/crossplane/blob/master/design/one-pager-managed-resource-api-design.md#high-fidelity).

Reply:
Thanks for the link! Went ahead and removed them both.